element-hq/element-x-android
# In-app language preferences

## Type of change


## Content

Following up on #2795 this PR adds a language preferences section in-app behind a feature flag.

There is no icon for it at the moment because there was no suitable icon within the design system.

There are 2 new strings that would need to be added to localazy.

I couldn't add tests on the presenter because of the access to context and I wasn't sure how to proceed with that.

## Motivation and context

Users of Android 12 or older would also like to change the language per-app and it turns out some OEMs are also hiding the system per-app language preferences on 13+.

## Screenshots / GIFs

|||
|-|-|
|![image](https://github.com/element-hq/element-x-android/assets/4409524/4493c94b-f276-434b-bde1-1fcda928c121)|![image](https://github.com/element-hq/element-x-android/assets/4409524/a64a6a91-ec31-4000-a28f-22236e06ecb0)|

## Tests



- Enable the feature flag in developer settings
- Open the new language section
- Test switching to different languages and back to default
- On >=13 check that the selection is also reflected on the system side
- On <=12 check that the language settings actually work and persist

## Tested devices

- OS version(s): 15 (beta 1.2), 12

## Checklist
